### PR TITLE
Legacy assembly formatter

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

**Comment by axic** [comment]
@chriseth please review

File: translate.js [hunk 1]
```diff
@@ -148,7 +148,41 @@ function translateJsonCompilerOutput (output) {
   return ret;
 }
 
+function formatAssemblyText (asm, prefix, source) {
+  if (typeof asm === typeof '' || asm === null || asm === undefined) {
+    return prefix + asm + '\n';
```
Comment (chriseth) [inline translate.js]: This will print `undefined` - do we want that?

File: translate.js [hunk 2]
```diff
@@ -148,7 +148,41 @@ function translateJsonCompilerOutput (output) {
...
+  }
+  var text = prefix + '.code\n';
+  asm['.code'].forEach(function (item, i) {
+    var v = item.value === undefined ? '' : item.value;
+    var src = '';
+    if (source !== undefined && item.begin !== undefined && item.end !== undefined) {
+      src = source.slice(item.begin, item.end).replace('\n', '\\n', 'g');
```
Comment (chriseth) [inline translate.js]: What about `\t`? Perhaps rather use some escaping function?

File: translate.js (hunk 1 above)
Comment (axic) [reply translate.js]: No. This was a verbatim copy from browser-solidity. I guess it should be `prefix + (asm || '') + '\n'`

File: translate.js (hunk 2 above)
Comment (axic) [reply translate.js]: True, fixed. Haven't found any built in escape function for those, do you know any?

**Comment by axic** [comment]
@chriseth ready for a second review

**Comment by axic** [comment]
@yann300 @iurimatias this can now be used in remix :)

**Pull request merged**